src/gui/DatabaseTabWidget.cpp

```diff
@@ -349,24 +304,31 @@ bool DatabaseTabWidget::closeAllDatabases()
     return true;
 }
 
-bool DatabaseTabWidget::saveDatabase(Database* db)
+bool DatabaseTabWidget::saveDatabase(Database* db, QString filePath)
 {
     DatabaseManagerStruct& dbStruct = m_dbList[db];
 
+    // Never allow saving a locked database; it causes corruption
+    Q_ASSERT(dbStruct.dbWidget->currentMode() != DatabaseWidget::LockedMode);
+    // Release build interlock
     if (dbStruct.dbWidget->currentMode() == DatabaseWidget::LockedMode) {
```

Comment:
This `if` won't be reached if true since the above assert will abort the execution. Better remove one of them